File: 404_SumofLeftLeaves.cpp
#include "DataStructure.h"

class Solution {
  public:
    int sumOfLeftLeaves(TreeNode* root) {
      if (!root) return 0;
      return sumOfLeafLeaves(root->left, true) + sumOfLeafLeaves(root->right, false);
    }

    int sumOfLeafLeaves(TreeNode* root, bool left) {
      if (!root) return 0;
      if (!root->left && !root->right)
        return left ? root->val : 0;
      return sumOfLeafLeaves(root->left, true) + sumOfLeafLeaves(root->right, false);
    }
};

int main() {
  Solution sol;
  return 0;
}

File: 404_SumofLeftLeaves.java
class SumOfLeftLeaves {
  public int sumOfLeftLeaves(TreeNode root) {
    if (root==null) return 0;
    return sumOfLeftLeaves(root.left, true) + sumOfLeftLeaves(root.right, false);
  }

  int sumOfLeftLeaves(TreeNode root, boolean left) {
    if (root==null) return 0;

    if (root.left==null && root.right==null && left)
      return root.val;
    else
      return sumOfLeftLeaves(root.left, true) + sumOfLeftLeaves(root.right, false);
  }
}